### PR TITLE
First pre-release

### PR DESCRIPTION
**Pull request by mexanick**
- Improved repo structure 
- added releasing tools 
- restructured some modules, notably optohybrid
- first testing RPC functions for configuration loading added

### CODE REVIEW

File: .github/uploads.cfg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{ "amc.so"                 :"/home/mdalchen/work/ctp7_modules/lib/amc.so",
+  "calibration_routines.so":"/home/mdalchen/work/ctp7_modules/lib/calibration_routines.so",
+  "extras.so"              :"/home/mdalchen/work/ctp7_modules/lib/extras.so",
+  "memory.so"              :"/home/mdalchen/work/ctp7_modules/lib/memory.so",
+  "optical.so"             :"/home/mdalchen/work/ctp7_modules/lib/optical.so",
+  "optohybrid.so"          :"/home/mdalchen/work/ctp7_modules/lib/optohybrid.so",
+  "rpctest.so"             :"/home/mdalchen/work/ctp7_modules/lib/rpctest.so",
+  "utils.so"               :"/home/mdalchen/work/ctp7_modules/lib/utils.so"}
```
Comment (jsturdy) [inline .github/uploads.cfg]: Again, question about hard coded paths

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@
 ./src/*.*~
 ./include/*.*~
 ./src/*.o
+./lib/*.so
```
Comment (jsturdy) [inline .gitignore]: Again comment about style for `.gitignore`

File: lib/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+*.so
```
Comment (jsturdy) [inline lib/.gitignore]: Usual comment about `.gitignore`

**Comment by mexanick** [comment]
Thanks for the comments, @jsturdy. I've addressed them, let me know if now it looks good to you.

**Review by jsturdy — APPROVED** [review]
Think it looks fine from my side

**Pull request merged**